### DISCUSSION TITLE
interpreting the data from Mlife

### DISCUSSION THREAD

**Discussion by lt72884** [OpenFAST/openfast#1777] [Q&A]
I have some questions regarding Mlife and understanding the data it gives me. Here are 2 snipets of data. The first is from an average wind speed of 12.5m/s and the second is from an average of 4.5m/s.

![image](https://github.com/OpenFAST/openfast/assets/52751207/5bc587dc-d51b-413e-b7d4-e01ce059106a)

Here is what i am trying to understand. At 12.5 m/s, the ultimate load is 806kN and will last roughly 20 years, but at a MUCH SLOWER wind speed, the ultimate load is drastically reduced to 280kN, but still lasts 20 years. This tells me that at slower wind speeds, the blade is weak and can only handle 280kN. This seems wrong because the ultimate load should be 806kN no matter the speed. 

IE, if the blades ultimate load is 806kN, then at 0 wind speed, the blades ultimate load should still be 806kN. SO why is the ultimate load so different?

nothing settings wise has changed at all besides changing the wind speed.

not sure if this helps or not, But for the wind speed of 12.5m/s, i would adjust L_ult for all 4 categories until the time was 20 years.

thanks

**Comment by jjonkman** [comment]
Dear @lt72884,

`L_Ult` is a user-specified value in MLife.  I agree that it should not depend on mean wind speed. But it is up the user of MLife to set the value of `L_Ult` correctly.

Best regards,

**Comment by lt72884** [comment]
see, thats the thing. I adjust the L_ult until the time hits 20 years. I adjust by using scaling factors stated in the FAQ post. So at 12.5m/s wind, the L_ult that had a life time of 20 years was 806kN.

but at 4m/s wind speed, the L_ult that allowed 20 years of life was extremely low at 280kN. Lower wind speeds should not effect the calculation. 

or maybe im understanding the data wrong? hahaha, thats possible too

**Reply by jjonkman** [reply]
`L_Ult` should not be calculated independently for different mean wind speed or load cases; `L_Ult` should be calculated universally.

The FAQ post you refer to discussed estimating `L_Ult` by scaling up the ultimate load computed via loads-analysis simulations.  This ultimate load should be the highest load across _all_ load case simulations.

Best regards,

**Comment by lt72884** [comment]
see, thats what i thought and whenever i put that value in, the time for life is around 726,000 years. Maybe i did it wrong. I used the average ultimate tensile strength of E-glass with 35% volume of fibers and 65% resin, which is what is used on the DOWEC 65 blade in the root sections and has a ultimate tensile strength (sigma) of near 600,000 psi. Since i know my cross sectional area of the root, i can then back out my force since sigma is F/A

i do not have FEA software available, and the faq said "If this (FEA) is not available—for conceptual designs for example—what we’ve done in the past is to see how the fatigue loads vary over a range of LUlt values. We’ve defined a range by taking the overall maximum value of the load in question (e.g., derived from MExtremes), and multiplying by scaling factors of 1.25, 2.5, 5, 10, and 20. The fatigue loads will asymptotically approach a fixed value for large values of LUlt."

so i scaled the 444kN which is the MExtreme value, by a number until the life time in seconds was 20 years

**Comment by lt72884** [comment]
in open fast, is it possible to change the angle of attack, pitch, initial twist of the blade, etc and how the blades behave during rotation? because i know all of these effect loading and also effect life time. 


thanks much

**Reply by jjonkman** [reply]
Yes, OpenFAST has all of these capabilities.

**Comment by lt72884** [comment]
ok, great. Which file do i need to edit for those? is that the elastadyn file?

thanks:)

**Reply by jjonkman** [reply]
The initial blade-pitch angle is set within ElastoDyn; if active control of pitch is enabled, this is set within ServoDyn or the controller selected within ServoDyn.

The aerodynamic twist is set within AeroDyn.

The angle of attack during rotation will be influenced by these and other factors, including the wind inflow and aero-elastic coupling.

Best regards

**Comment by lt72884** [comment]
perfect, thank you much for that.

An odd question. I am trying to find what is called ix,iy,iz, (capital i) which is part of the moment equation of Mct/Ix. I would like to find Ix, Iy and Iz of each section of the blade since the blade has been divided into sections. Will openfast output show this?

i want to graph each "I" and then curve fit it to find an equation. This is for a paper for a report for school. 

thanks

**Reply by jjonkman** [reply]
I'm not sure I understand you question.  Are you referring to the cross-sectional area moments of inertia?

**Comment by lt72884** [comment]
Yeah, i dont blame you, its a hard question to type. its the bending stress where Sigma= Mc/I where M is the moment, c is the centroid distance, and I is the moment of inertia. Does openFast find values of I? 

thanks

**Reply by jjonkman** [reply]
OpenFAST does not make use of the second area moments of inertia of the the blade cross sections.  The blade bending stiffness at each cross section (in flap and edge) is input in ElastoDyn, and in BeamDyn, similar stiffness terms as well as the second mass moments of inertia are input at each cross section.  If the blade was composed of a simple isotropic material, then these cross-sectional bending stiffness and cross-sectional second mass moments of inertia would be proportional to the second area moments of inertia, but blades are typically made of composite materials where those simple relations don't apply.

**Comment by lt72884** [comment]
is there a way to add shear or youngs modulus for material properties to any of the .dat files for the 5mw_land_dll_wturb folder? i would like to test it out on this particular blade:) thanks

**Reply by jjonkman** [reply]
Dear @lt72884,

If you are modeling blades in ElastoDyn, the bending stiffness of the blade at each cross section in the flapwise and edgewise directions is specified by the user in the ElastoDyn blade data file.  These bending stiffness terms would naturally be based on the Young's modulus, but would also include the cross-sectional second area moment of inertia as well.  The Shear modulus is not directly used by ElastoDyn, which considers blade bending, but neglects blade shear and torsion.

If you are modeling blades in BeamDyn, the full 6x6 cross-sectional stiffness matrix is input at each cross section.  The bending stiffness specified for ElastoDyn are elements of this full matrix, but the full matrix also includes shear terms, torsion terms, stretching terms, and couplings.  So, the Young's modulus as well as the Shear modulus would be needed to populate this stiffness matrix, but other properties of the cross section are needed as well.

Best regards,

**Comment by lt72884** [comment]
ok, great. ill take a look at the elstadyn.dat file and see if i can find that section. 

What is the default value for the cross sectional second moment? this way i can solve for other parameters such as area in case i need it. 

thanks

**Reply by jjonkman** [reply]
Neither ElastoDyn nor BeamDyn have default values for the cross-sectional shape.  Neither the modulus nor the geometry of the cross section are specified directly in (or assumed by) ElastoDyn or BeamDyn.  Rather, elements of the cross-sectional stiffness matrix are set within ElastoDyn and BeamDyn.

Best regards,

**Comment by lt72884** [comment]
ok, so if i wanted to add my own youngs modulous to the elastadynblade1.dat file, is that possible so i can test out some material options?

thank you my friend :)

**Reply by jjonkman** [reply]
You can specify you own blade stiffness properties in ElasoDyn, though not the Young's Modulus directly (as I mentioned above).

**Comment by lt72884** [comment]
Ok, so do i use the "NRELOffshrBsline5MW_Onshore_ElastoDyn_BDoutputs.dat" found in the 5mw-land-dll-wturb folder or am i using the 
"NRELOffshrBsline5MW_Blade.dat" in the 5mw-baseline folder to specify blade stiffness for elstodyn?

thanks

**Reply by jjonkman** [reply]
The ElastoDyn blade stiffness for the NREL 5-MW baseline wind turbine is specified in the _NRELOffshoreBsline5MW_Blade.dat_ file.

**Reply by lt72884** [reply]
ok, and this file is used for the 5mw-land-dll-wturb wind turbine correct?

i noticed there are 50 entries, do i need to edit all 50 of those? if so where do i find the second moments so i can do EI/l^3 

thanks

**Comment by lt72884** [comment]
here are the current values i have. How are these calculated? IE 1.8113600E+10 (first value of edge stiffness) came from what equation and what were the values for each variable? 

thank you :)

https://photos.app.goo.gl/yB2f7wj2Eo2pkJmM6

**Reply by jjonkman** [reply]
Yes, _NRELOffshoreBsline5MW_Blade.dat_ is used by the _5MW_Land_DLL_WTurb.fst_ r-test model.  You can track which input files are used by given a model through the input files themselves.  The OpenFAST primary (_.fst_) file lists which module-level input files it uses and some module-level inputs (such as ElastoDyn) list additional data files used by that module (e.g., the ElastoDyn input file identifies blade and tower input files).

Table 10 from the old FAST v6 User's Guide (https://openfast.readthedocs.io/en/main/_downloads/d8bd014121d6505cb25cf49bee5eaa80/Old_FAST6_UsersGuide.pdf) describes what `FlpStff` and `EdgStff` in the ElastoDyn blade input file are.  For a simple cross section with an isotropic material, the stiffness values are `FlpStff` = E * I_xx and `EdgStff` = E * I_yy.

Best regards,

**Comment by lt72884** [comment]
ok, great. I have been reading that the last few minutes. So if i wanted to find out what what value of E they are using, is there a way to find the I_xx and I_yy values so i can back out E so i can find out what material is being used? IE, 1.8113600E+10 = E*I_xx

lastly, what equation is openfast using for I_xx and I_yy?

I am using the 5mw-land-dll-wturb

thank you for doing this for me :)

**Reply by jjonkman** [reply]
The simplification FlpStff = E * I_xx and EdgStff = E * I_yy is stated to aid understanding.  OpenFAST does use not these specific questions.  In reality, a sectional analysis tool would be used to derive these bending stiffness for a complex blade cross section composed of composite material and airfoil shape.

**Comment by lt72884** [comment]
so how did all the values for flpstff and edgstff get into the elstadyn blade file? 

Thanks

**Reply by jjonkman** [reply]
For the NREL 5-MW baseline wind turbine, the blade-bending stiffness values were taken from the DOWEC study, as described in the specifications report of the NREL 5-MW baseline wind turbine: https://www.nrel.gov/docs/fy09osti/38060.pdf.

**Comment by sjihf12** [comment]
Hello , Dear Professor Jjonkman, can you kindly answer some questions about MLIFE for me?
1.Can MLIFE calculate fatigue life of a mooring line ? If not, which software can calculate the  fatigue life of a mooring line?
2.If MLIFE's 'DesignLife' is set to 630720000? Should my input file also provide data of length 630720000? Why?

**Reply by jjonkman** [reply]
Dear @sjihf12,

Here are my responses:

1. MLife is a general tool for post-processing fatigue based on rainflow-cycle counting.  It can apply equally to all components of a wind turbine and offshore support structure.
2. No.  MLife will extrapolate the fatigue cycles based on the short-term data provided as input to MLife (typically a set of 10-minute long load-case simulations) and the probability of occurrence of a particular simulation (i.e., binned by mean wind speed) in order to estimate the fatigue over 20-years (630720000 s).

Best regards,

**Comment by lt72884** [comment]
thank you for the reply. i will read that today and see what i find.

i noticed in mlife, the woehler exponent is set to 10. From what i have read, the higher the better, but not to go over 10? is this true?

**Comment by lt72884** [comment]
in the "NRELOffshrBsline5MW_Onshore_ElastoDyn_BDoutputs" I have a section labeled "initial conditions" as shown below. I am trying to find what the appropriate ranges are for EACH value? IE, BlPitch(1) should be what value? and what should it NOT go above or below? what is a realistic value for each one of these? or at least a range that i can work with

thanks

the docs only list what they mean, but no ranges i should fall inbetween

https://raf-openfast.readthedocs.io/en/docs-fast8readme/source/user/elastodyn/input.html#initial-conditions

---------------- INITIAL CONDITIONS --------------------------------------
          0   OoPDefl     - Initial out-of-plane blade-tip displacement (meters)
          0   IPDefl      - Initial in-plane blade-tip deflection (meters)
          0   BlPitch(1)  - Blade 1 initial pitch (degrees)
          0   BlPitch(2)  - Blade 2 initial pitch (degrees)
          0   BlPitch(3)  - Blade 3 initial pitch (degrees) [unused for 2 blades]
          0   TeetDefl    - Initial or fixed teeter angle (degrees) [unused for 3 blades]
          0   Azimuth     - Initial azimuth angle for blade 1 (degrees)
       12.1   RotSpeed    - Initial or fixed rotor speed (rpm)
          0   NacYaw      - Initial or fixed nacelle-yaw angle (degrees)
          0   TTDspFA     - Initial fore-aft tower-top displacement (meters)
          0   TTDspSS     - Initial side-to-side tower-top displacement (meters)
          0   PtfmSurge   - Initial or fixed horizontal surge translational displacement of platform (meters)
          0   PtfmSway    - Initial or fixed horizontal sway translational displacement of platform (meters)
          0   PtfmHeave   - Initial or fixed vertical heave translational displacement of platform (meters)
          0   PtfmRoll    - Initial or fixed roll tilt rotational displacement of platform (degrees)
          0   PtfmPitch   - Initial or fixed pitch tilt rotational displacement of platform (degrees)
          0   PtfmYaw     - Initial or fixed yaw rotational displacement of platform (degrees)

**Reply by jjonkman** [reply]
To minimize start-up transients and to avoid initialization issues with the controller, we typically recommend to set `RotSpeed` and `BlPitch` to their expected (mean) values conditioned on each wind speed you are simulating.  You will need to know the expected steady-state response to know those.  For the NREL 5-MW baseline wind turbine, you can find those in Figure 9-1 of its specifications report: https://www.nrel.gov/docs/fy09osti/38060.pdf.

For floating offshore wind turbines (which it doesn't look like you are simulating), we also recommend to minimize start-up transients that you set `PtfmSurge` and `PtfmPitch` to their expected (mean) values conditioned on each wind speed you are simulating.

The remaining values can also be set based on their expected (mean) values, but they are generally less critical to set and can generally be left at zero unless you want them nonzero for some specific case, e.g., a free-decay simulation.

Best regards,

**Comment by lt72884** [comment]
i dont understand your reply :) figure 9-1 is a list of frequencies. Blade pitch is in degrees and rotational speed is RPM

thanks

**Reply by jjonkman** [reply]
Table 9-1 lists the frequencies; Figure 9-1 shows the steady-state solution in the NREL 5-MW specifications report.

**Comment by lt72884** [comment]
i am so sorry i got that confused. thank you for showing me that. I work the swing shift at a treatment center for youth with drug and alcohol addiction as their emotional guidance mentor/consoler and i was exhausted from running some one on ones.

**Comment by lt72884** [comment]
if i wanted to create the same graphs as figure 9-1, how would i do that? did the author(you) just run many many tests and then compile the data? IE, for the blade pitch line, why keep it at 0 for slow speeds? do you lose power generation? and how did they know that at 12.5m/s the blade pitch was about 4.5 degrees? I know the paragraph before 9-1 speaks briefly about it

if i wanted to create that exact graph, how is the best way to do this? I feel this may help me learn something new

thanks

**Reply by jjonkman** [reply]
Dear @lt72884,

As described in the NREL 5-MW specifications report, we generated the results of Figure 9-1 by running a series of FAST with AeroDyn simulations at a number of given, steady, and uniform wind speeds. Separate simulations were run at each wind speed and each simulation was run long enough to ensure that all transient behavior had died out. We recorded the steady-state output values from each simulation and plotted them as a line in Figure 9-1 (thus, only the steady-state response is shown).

Blade-pitch should remain constant below rated to maximize power capture (to ensure you are operating at peak efficiency).  The pitch angle above rated comes from the controller to maintain the rotor at rated speed.

Best regards,

**Comment by lt72884** [comment]
thanks for this information. i really apreciate it

**Comment by lt72884** [comment]
ok, so i did some basic tests of my own and wrote down the values. i created many files with different pitch angels but the same wind file. i compared the angle, max loading and power generation and for my results, i noticed that power generation stays constant until the blade angle hit roughly 25 degrees. 

the run time was set for 30 minutes

here are my outputs. 

https://photos.app.goo.gl/vhQpgXpsSBT9j1qD8

**Reply by jjonkman** [reply]
I'm not understanding.  By "different pitch angles", do you mean you changed the initial conditions of blade pitch or you disabled the controller and used fixed pitch angles instead?

**Comment by lt72884** [comment]
i went into the blade.dat file and changed it from 0 to different angles to see how the other properties such as loading would differ.

thanks
